src/function/mod.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{
+    error::{FreightError, OrReturn},
+    execution_engine::{evaluate, ExecutionEngine},
+    expression::Expression,
+    TypeSystem,
+};
+use std::fmt::Debug;
+
+mod arg_count;
+mod function_ref;
+mod function_type;
+mod function_writer;
+
+pub use arg_count::*;
+pub use function_ref::*;
+pub use function_type::*;
+pub use function_writer::*;
+
+#[derive(Debug)]
+pub struct Function<TS: TypeSystem> {
+    pub(crate) expressions: Vec<Expression<TS>>,
+
+    pub(crate) _stack_size: usize,
+    pub(crate) arg_count: ArgCount,
+    pub(crate) return_target: usize,
+}
+
+impl<TS: TypeSystem> Function<TS> {
+    pub fn call(
+        &self,
+        engine: &mut ExecutionEngine<TS>,
+        args: &mut [TS::Value],
+        captured: &[TS::Value],
+        // real:usize
+    ) -> Result<TS::Value, FreightError> {
+        match self.arg_count.max() {
+            Some(max) if max > args.len() => {
```

Comment:
Shouldn't this be `if args.len() > max`? It seems like this is returning an error if they provided *less* than the maximum

Reply:
this shouldn't be there at all, its supposed to be a call to a function of ArgCount but I guess that got deleted
so Ill put that back in